src/Toolchain/ToolchainProvider.ts

```diff
@@ -109,138 +109,164 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
       new vscode.EventEmitter<ToolchainTreeData>();
   readonly onDidChangeTreeData?: vscode.Event<ToolchainTreeData> = this._onDidChangeTreeData.event;
 
-  constructor() {}
-
-  getTreeItem(element: BaseNode): vscode.TreeItem {
+  public getTreeItem(element: BaseNode): vscode.TreeItem {
     return element;
   }
 
-  getChildren(element?: BaseNode): Thenable<BaseNode[]> {
+  public getChildren(element?: BaseNode): Thenable<BaseNode[]> {
     if (element === undefined) {
       return Promise.resolve(NodeBuilder.createBackendNodes());
     } else {
       return Promise.resolve(NodeBuilder.createToolchainNodes(element));
     }
   }
 
-  error(msg: string, ...args: string[]): Thenable<string|undefined> {
+  /* istanbul ignore next */
+  private error(msg: string, ...args: string[]): Thenable<string|undefined> {
     Logger.error(this.tag, msg);
     return vscode.window.showErrorMessage(msg, ...args);
   }
 
-  refresh() {
+  public refresh() {
```

Comment:
Maybe this can be ignored too?

Reply:
Could you explain your intention a little more?

Reply:
I thought that this function is about UI but my misunderstanding, sorry T.T